vaccine_feed_ingest/runners/us/physicians_immediate/parse.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+import json
+import pathlib
+import re
+import sys
+from typing import Dict, List
+
+from bs4 import BeautifulSoup
+
+ALLOWED_NORMALIZED_COLUMNS = {"clinic", "slots", "type", "address", "hours"}
+
+
+def soupify_file(input_path: pathlib.Path) -> List[BeautifulSoup]:
+    """Opens up a provided file path, feeds it into BeautifulSoup.
+    Returns a new BeautifulSoup object for each found table
+    """
+    with open(input_path, "r") as fd:
+        return BeautifulSoup(fd, "html.parser").find_all("table")
+
+
+def find_data_rows(soup: BeautifulSoup) -> List[BeautifulSoup]:
+    """Queries the provided BeautifulSoup to find <tr> elements which are inside a <tbody>.
+    Exploring the data shows that such rows correspond to vaccine site data
+    """
+
+    def is_data_row(tag):
+        return tag.name == "tr" and tag.parent.name == "tbody"
+
+    return soup.find_all(is_data_row)
+
+
+def find_column_headings(soup: BeautifulSoup) -> List[str]:
+    """Queries the provided BeautifulSoup to find <th> elements under a <thead>.
+    Column names are normalized for later conditional parsing.
+    Returns a list of normalized column names
+    """
+
+    def translate(col: str) -> str:
+        """The source HTML isn't always consistent in how columns are named.
+        This function allows us to map divergent labels onto the correct label for parsing
+        """
+        translation = {"types": "type", "name": "clinic"}
+        if col in translation.keys():
+            return translation[col]
+        else:
+            return col
+
+    headings = [th.contents for th in soup.find_all("th") if len(th.contents) > 0]
+    normalized_cols = ["_".join(col).lower() for col in headings]
+    translated_cols = map(translate, normalized_cols)
+    return [col for col in translated_cols if col in ALLOWED_NORMALIZED_COLUMNS]
+
+
+def parse_row(row: BeautifulSoup, columns: List[str]) -> Dict[str, str]:
+    """Takes a BeautifulSoup tag corresponding to a single row in an HTML table as input,
+    along with an ordered list of normalized column names.
+    Labels data in each row according to the position of the column names.
+    Returns a dict of labeled data, suitable for transformation into ndjson
+    """
+
+    def extract_appt_slot_count(appt_slots: str) -> str:
+        pattern = re.compile("(\d+) slots")
```

Comment:
Flake8 complains about this line
> W605 invalid escape sequence '\d'

You can fix this by making this a raw string with a `r` prefix

```suggestion
        pattern = re.compile(r"(\d+) slots")
```